src/conditions/base/contract.ts

```diff
@@ -6,31 +7,51 @@ import { RpcCondition, rpcConditionRecord } from './rpc';
 
 export const STANDARD_CONTRACT_TYPES = ['ERC20', 'ERC721'];
 
-const functionAbiVariable = Joi.object({
-  internalType: Joi.string(), // TODO is this needed?
-  name: Joi.string().required(),
-  type: Joi.string().required(),
-});
-
 const functionAbiSchema = Joi.object({
   name: Joi.string().required(),
   type: Joi.string().valid('function').required(),
-  inputs: Joi.array().items(functionAbiVariable),
-  outputs: Joi.array().items(functionAbiVariable),
-  // TODO: Should we restrict this to 'view'?
-  stateMutability: Joi.string(),
+  inputs: Joi.array(),
+  outputs: Joi.array(),
+  stateMutability: Joi.string().valid('view', 'pure').required(),
 }).custom((functionAbi, helper) => {
+  // Is `functionABI` a valid function fragment?
+  let asInterface;
+  try {
+    asInterface = new ethers.utils.Interface([functionAbi]);
+  } catch (e: unknown) {
+    const { message } = e as Error;
+    return helper.message({
+      custom: message,
+    });
+  }
+
+  if (!asInterface.fragments) {
+    return helper.message({
+      custom: '"functionAbi" is missing a function fragment',
+    });
+  }
+
+  if (asInterface.fragments.length > 1) {
+    return helper.message({
+      custom: '"functionAbi" must contain exactly one function fragment',
+    });
+  }
+
+  // Now we just need to validate against the parent schema
   // Validate method name
   const method = helper.state.ancestors[0].method;
-  if (functionAbi.name !== method) {
+  const functionFragment = asInterface.fragments.filter(
+    (f) => f.name === method
+  )[0];
+  if (!functionFragment) {
     return helper.message({
-      custom: '"method" must be the same as "functionAbi.name"',
+      custom: '"functionAbi" does not contain the method specified as "method"',
```

Comment:
Can we output the actual method name. So something like (?):
```suggestion
      custom: `"functionAbi" not valid for method, "${method}"`,
```